ui/admin/app/routes/scopes/scope/credential-stores/credential-store/credential-libraries/credential-library.js

```diff
@@ -19,4 +20,28 @@ export default class ScopesScopeCredentialStoresCredentialStoreCredentialLibrari
       reload: true,
     });
   }
+
+  /**
+   * Copies the contents of array fields in order to force the instance
+   * into a dirty state.  This ensures that `model.rollbackAttributes()` reverts
+   * to the original expected array.
+   *
+   * The deep copy implemented here is required to ensure that both the
+   * array itself and its members are all new.
+   *
+   * @param {credentialLibraryModel} credentialLibrary
+   */
+  @action
+  edit(credentialLibrary) {
+    if (credentialLibrary.critical_options) {
+      credentialLibrary.critical_options = structuredClone(
+        credentialLibrary.critical_options
+      );
+    }
+    if (credentialLibrary.extensions) {
+      credentialLibrary.extensions = structuredClone(
+        credentialLibrary.extensions
+      );
+    }
```

Comment:
`structuredClone` works with `undefined` and `null`, so it should be safe to do unguarded.
```suggestion
    const { critical_options, extensions } = credentialLibrary;
    credentialLibrary.critical_options = structuredClone(critical_options);
    credentialLibrary.extensions = structuredClone(extensions);
```